gaztec/src/components/contract/components/registerContractDialog.tsx

```diff
@@ -0,0 +1,124 @@
+import DialogTitle from "@mui/material/DialogTitle";
+import Dialog from "@mui/material/Dialog";
+import { Contract } from "@aztec/aztec.js";
+import {
+  Button,
+  CircularProgress,
+  FormControl,
+  FormGroup,
+  TextField,
+  Typography,
+  css,
+} from "@mui/material";
+import { useContext, useState } from "react";
+import { AztecAddress, ContractInstanceWithAddress } from "@aztec/circuits.js";
```

Comment:
`aztec.js` exports `AztecAddress` as well, should we use that one? do we need circuits? is it all circular imports?

Reply:
It's all circular imports all the way down. Making imports tidy is next on the TODO list, and `circuit-types` vs `circuits.js` + all the reexports definitely scares me...